File: Beelina.APP/src/app/_models/product-withdrawal-entry.ts
import { IModelNode } from '../_interfaces/imodel-node';
import { Entity } from './entity.model';
import { ProductStockAudit } from './product-stock-audit';
import { User } from './user.model';
import { NumberFormatter } from '../_helpers/formatters/number-formatter.helper';

export class ProductWithdrawalEntry extends Entity implements IModelNode {
  public userAccountId: number;
  public userAccount: User;
  public stockEntryDate: Date;
  public withdrawalSlipNo: string;
  public notes: string;
  public productStockAudits: Array<ProductStockAudit>;
  public totalAmount: number;

  get formattedTotalAmount(): string {
    return NumberFormatter.formatCurrency(this.totalAmount || 0);
  }

  constructor() {
    super();
    this.userAccount = new User();
    this.productStockAudits = [];
    this.totalAmount = 0;
  }
}

File: Beelina.LIB/Models/ProductWithdrawalEntry.cs
using Beelina.LIB.Interfaces;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;

namespace Beelina.LIB.Models
{
    public class ProductWithdrawalEntry
    : Entity, IUserActionTracker
    {
        public int? UserAccountId { get; set; }
        public DateTime? StockEntryDate { get; set; }
        public string WithdrawalSlipNo { get; set; }
        public string Notes { get; set; }

        public List<ProductStockAudit> ProductStockAudits { get; set; }

        public virtual UserAccount UserAccount { get; set; }

        [NotMapped]
        public double TotalAmount { get; set; }

        [NotMapped]
        public string FormattedTotalAmount
        {
            get
            {
                return TotalAmount.ToString("C", new CultureInfo("en-PH"));
            }
        }

        public int? DeletedById { get; set; }
        public virtual UserAccount DeletedBy { get; set; }
        public int? UpdatedById { get; set; }
        public virtual UserAccount UpdatedBy { get; set; }
        public int? CreatedById { get; set; }
        public virtual UserAccount CreatedBy { get; set; }
        public int? DeactivatedById { get; set; }
        public virtual UserAccount DeactivatedBy { get; set; }
    }
}